OpenSim/Common/DelimFileAdapter.h

```diff
@@ -370,10 +371,17 @@ DelimFileAdapter<T>::extendRead(const std::string& fileName) const {
 
     // Read the line containing column labels and fill up the column labels
     // container.
-    auto column_labels = nextLine();
+    std::vector<std::string> column_labels{};
+    while (column_labels.size() == 0) { // keep going down rows to find labels
+        column_labels = nextLine();
+        // for labels we never expect empty elements, so remove them
+        IO::eraseEmptyElements(column_labels);
+        ++line_num;
+    }
```

Comment:
If I don't do this I cannot distinguish a line with blanks from a line with labels, since I will interpret the line to be empty labels. We would have  to make our format stricter to say there cannot be blank lines between the header and the data, but I believe we have files that do have empty lines, or some random amount of tabs or spaces.

Reply:
Thank you for explaining. However, if `no-empty-column-labels` is an invariant, we should make sure it's enforced properly (so that we can read files we write), and we should also document this.

Do you think it's more likely for files out in the wild to (A) have empty lines between `endheader` and the column labels or (B) have a few empty column labels?

Reply:
I have seen cases of (A) and none for (B). I don't believe the previous parsing machinery was capable of processing B since contiguous delimiters  were being treated as one.

I have added enforcement to DataTable such that column labels cannot be empty and cannot contain delimiter characters or have leading and trailing spaces.